pkg/corerp/api/v20231001preview/environment_conversion.go

```diff
@@ -151,6 +170,27 @@ func (dst *EnvironmentResource) ConvertFrom(src v1.DataModelInterface) error {
 		dst.Properties.Recipes = recipes
 	}
 
+	if !reflect.DeepEqual(env.Properties.RecipeConfig, datamodel.RecipeConfigProperties{}) {
+		recipeConfig := &RecipeConfigProperties{}
+		if !reflect.DeepEqual(env.Properties.RecipeConfig.Terraform, datamodel.TerraformConfigProperties{}) {
```

Comment:
Same as above, since this logic is specific to Terraform, maybe we can move them to a helper function

Reply:
We can apply the same thing as I mentioned above but this time in reverse.

**This function -> RecipeConfig conversion function -> Terraform related conversion function**

In future, it would be very easy to add new types of provider conversions.

Reply:
created a separate func

Reply:
I agree with creating a Terraform related conversion function. @vishwahiremat I know you have updated it to recipe config specific conversion, have you considered adding Terraform specific function to make it easy to extend the recipe config function in the future?

Reply:
Vishwa and I discussed offline, we will abstract it out into Terraform specific function once we add support for other types.